test/run

```diff
@@ -0,0 +1,268 @@
+#!/bin/bash -x
+#
+# The 'run' performs a simple test that verifies that S2I image.
+# The main focus here is to excersise the S2I scripts.
+#
+# IMAGE_NAME specifies a name of the candidate image used for testing.
+# The image has to be available before this script is executed.
+#
+IMAGE_NAME=${IMAGE_NAME-quay.io/centos7/ruby-27-centos7-candidate}
+
+declare -a WEB_SERVERS=(db puma rack)
+#declare -a WEB_SERVERS=(db)
+
+# TODO: Make command compatible for Mac users
+test_dir="$(readlink -zf $(dirname "${BASH_SOURCE[0]}"))"
+image_dir=$(readlink -zf ${test_dir}/..)
+test_short_summary=''
+TESTSUITE_RESULT=0
+
+TEST_LIST="\
+test_docker_run_usage
+test_application
+test_connection
+test_scl_variables_in_dockerfile
+test_scl_usage
+test_npm_functionality
+"
+source "${test_dir}/test-lib.sh"
+
+# Read exposed port from image meta data
+test_port="$(docker inspect --format='{{range $key, $value := .Config.ExposedPorts }}{{$key}}{{end}}' ${IMAGE_NAME} | sed 's/\/.*//')"
+
+info() {
+  echo -e "\n\e[1m[INFO] $@...\e[0m\n"
+}
+
+image_exists() {
+  docker inspect $1 &>/dev/null
+}
+
+container_exists() {
+  image_exists $(cat $cid_file)
+}
+
+container_ip() {
+  docker inspect --format="{{ .NetworkSettings.IPAddress }}" $(cat $cid_file)
+}
+
+run_s2i_build() {
+  ct_s2i_build_as_df file://${test_dir}/${1}-test-app ${IMAGE_NAME} ${IMAGE_NAME}-testapp ${s2i_args} $(ct_build_s2i_npm_variables)
+}
+
+run_test_application() {
+  docker run --user=100001 --rm --cidfile=${cid_file} ${IMAGE_NAME}-testapp
+}
+
+cleanup() {
+  local server="$1"
+  info "Cleaning up the test application image $1"
+  if [ -f $cid_file ]; then
+    if container_exists; then
+      docker stop $(cat $cid_file)
+    fi
+  fi
+  if image_exists ${IMAGE_NAME}-testapp; then
+    docker rmi -f ${IMAGE_NAME}-testapp
+  fi
+  if [ ! -z "${server}" ]; then
+    rm -rf ${test_dir}/${server}-test-app/.git
+  fi
+  if [[ "${server}" == "db" ]]; then
+    rm -rf ${test_dir}/db-test-app
+  fi
+}
+
+check_result() {
+  local result="$1"
+  TESTCASE_RESULT=0
```

Comment:
Each `check_result` reset the value. If in some test case is called two times, then each `check_result` resets it, which is wrong.

Reply:
Hm, this is true, good catch. Going to fix this.

Reply:
Thanks for the commit suggestions.